Base/res/js/Spreadsheet/runtime.js

```diff
@@ -57,6 +57,56 @@ class Position {
         );
     }
 
+    range_up() {
+        if (this.row === 0) throw new Error(`No cells above this cell`);
+        const up_one = this.up(1);
+        let current_point = up_one;
+        for (
+            let point = current_point.up(1);
+            current_point.row !== 0 && typeof point.value() === "number";
+            point = current_point.up(1)
+        )
+            current_point = point;
+        return range(current_point.name, up_one.name);
+    }
+
+    range_down() {
+        let down_one = this.down(1);
+        let current_point = down_one;
+        for (
```

Comment:
This boi will just keep going forever in something like `A1: =here().range_down()`, I can't quite remember if referencing cells outside the currently generated area will expand it or not, so it might _actually_ hang forever

Reply:
Doesn't hang, just tested it. I put values in A1, A2, and A3. Then I put that formula in A0, then watched it complete quickly as expected, and nothing hung :)

I also tested with just a completely empty sheet and that formula in the top left, and it completed as expected and nothing hung.

Reply:
![J9XCFtP](https://user-images.githubusercontent.com/43508353/154188400-cb825d8c-1893-4daf-8b26-c27820a60329.png)
![GbJFLKV](https://user-images.githubusercontent.com/43508353/154188443-c7f92e03-c20b-4877-b70b-bf428a5a3e21.png)